rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -142,6 +228,23 @@ wait(
     }
   }
 
+  std::unordered_set<DDS::StatusCondition *> status_conditions;
+  // gather all status conditions with set masks
+  rmw_ret_t ret_code = __gather_event_conditions(events, status_conditions);
+  if (ret_code != RMW_RET_OK) {
+    return ret_code;
+  }
+  if (!status_conditions.empty()) {
```

Comment:
Is this check needed? If it is empty, it looks like nothing will happen anyways.

Reply:
There is a difference between an empty set and the a ! RMW_RET_OK status: __gather_event_conditions will set the relevant error message and return RMW_RET_ERROR for a few different reasons.

Reply:
Agreed